### PR TITLE
Adds Badge component docs to site

### PR DESCRIPTION
**Pull request by james-nash**
Adds the Badge component docs from the completed Word template to the site

### Notes for reviewers:

* You can preview this page here: https://saltdesignsystem-git-add-badge-docs-joshwooding.vercel.app/salt/components/badge/usage
* This PR only converts the text content from Word to MDX. The content has already been reviewed in Word, so you just need to focus your review on whether or not it's been added to the site correctly.
* The example code still needs to be migrated out of Storybook, so you will just see a dummy example on the page for now (but the titles & descriptions for each example should already be correct)
* The props table will appear empty because it's expecting to find the code in the `core` package, but this component is still in `labs`. I figured it's better to do it this way around rather than setting it to `labs` now and risking forgetting to update that before this page goes live to consumers. (I did test it with labs though and the props appears correctly, so this _will_ work when the time comes)
* I'm aware that the Accessibility tab needs some spacing between the tabs and the text. However, this is a bug in the page styling and nothing to do with this component's content. I'll therefore address that in a separate PR

### CODE REVIEW

File: site/docs/components/badge/examples.mdx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+<LivePreview componentName="component-name" exampleName="ComponentExample" title="Badge with string">
+
+`value` can accept both number and string.
+
+</LivePreview>
+
+<LivePreview componentName="component-name" exampleName="ComponentExample" title="Badge with truncated string">
+
+When a string of greater than 4 characters is passed to `value` it is truncated and appended with `...`.
```
Comment (lukeac123) [inline site/docs/components/badge/examples.mdx:38]: this truncation has also been removed - same as above will change and push put a new commit

**Review by lukeac123 — COMMENTED** [review]
Badge characteristics have changed since this was agreed by content. Will push up a new commit with updated characteristics

**Review by lukeac123 — APPROVED** [review]

File: site/docs/components/badge/usage.mdx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+layout: DetailComponent
+sidebar:
+  label: Usage
+data:
+  $ref: ./#/data
+---
+
+### Using the Badge component
+
+#### When not to use Component Name
```
Comment (feliciackh) [inline site/docs/components/badge/usage.mdx:14]:
```suggestion
#### When not to use Badge
```

File: site/docs/components/badge/usage.mdx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+```js
+import { Badge } from "@salt-ds/core";
+```
+
+### Props
+
+{/* TODO: Update this to grab from core package when component is released */}
+
+<PropsTable packageName="labs" componentName="Badge" />
```
Comment (feliciackh) [inline site/docs/components/badge/usage.mdx:33]:
```suggestion
<PropsTable packageName="core" componentName="Badge" />
```
might be a good idea to have it as packageName="core" (in alignment with what's mentioned above in Import section, and all other component docs)

**Comment by origami-z** [comment]
Will be covered by a new PR by #2199

**Pull request closed without merging**